pubtools/_quay/operator_pusher.py

```diff
@@ -453,25 +453,95 @@ def build_index_images(self):
 
         """
         iib_results = {}
+        repos_opted_in = {}
+        items_opted_in = {}
+        failed_items = {}
+
+        # We need to load pyxis resolved versions at this point
+
+        versions_mapping = self.version_items_mapping  # noqa: F841
+
+        for item in self.push_items:
+            for repo in item.metadata["tags"]:
+                if repo not in repos_opted_in:
+                    repos_opted_in[repo] = pyxis_get_repo_metadata(repo, self.target_settings).get(
+                        "fbc_opt_in", False
+                    )
+            item_fbc_opt_in = [repos_opted_in[repo] for repo in item.metadata["tags"]]
+            if not (all(item_fbc_opt_in) or not any(item_fbc_opt_in)):
+                failed_items[id(item)] = True
+                item.add_error("NOTPUSHED", "Inconsistent fbc opt in")
```

Comment:
What do you mean?

Reply:
I suppose push items will end up with PENDING rather than NOTPUSHED or INVALIDFILE if update_push_items is not called. The only calling I can find is https://github.com/release-engineering/pubtools-quay/blob/master/pubtools/_quay/push_docker2.py#L42, but it's obsolete code.

Reply:
I believe there's pushcollector call update_push_items in pub after target operation finishes

Reply:
Yeah, but there's sys.exit(1) after tags rollback so update_push_items in pub won't be run.